packages/webview_flutter/test/webview_flutter_test.dart

```diff
@@ -115,20 +287,44 @@ class FakePlatformWebView {
 
   MethodChannel channel;
 
-  String lastUrlLoaded;
+  List<String> history = <String>[];
+  int currentPosition = -1;
+  String get lastUrlLoaded =>
+      (currentPosition > -1 && history.length > currentPosition)
+          ? history[currentPosition]
+          : null;
   JavaScriptMode javaScriptMode;
 
   Future<dynamic> onMethodCall(MethodCall call) {
     switch (call.method) {
       case 'loadUrl':
-        lastUrlLoaded = call.arguments;
+        final String url = call.arguments;
+        if (history.length - 1 > currentPosition) {
```

Comment:
nit: IIUC the `sublist` call below should always be safe do I'd just drop this condition for readability.